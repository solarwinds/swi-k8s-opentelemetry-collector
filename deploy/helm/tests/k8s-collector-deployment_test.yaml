# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for k8s-collector-deployment
templates:
  - network/k8s-collector-deployment.yaml
  - network/configmap.yaml
tests:
  - it: K8s Collector spec should match snapshot when using default values
    template: network/k8s-collector-deployment.yaml
    set:
      ebpfNetworkMonitoring.enabled: true
    asserts:
      - matchSnapshot:
          path: spec.template.spec