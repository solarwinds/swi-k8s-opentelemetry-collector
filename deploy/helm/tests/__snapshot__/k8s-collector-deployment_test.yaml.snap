K8s Collector spec should match snapshot when using default values:
  1: |
    containers:
      - args:
          - --log-console
          - --log-level=warning
        image: otel/opentelemetry-ebpf-k8s-watcher:v0.10.1
        imagePullPolicy: IfNotPresent
        name: k8s-watcher
      - args:
          - --config-file=/etc/network-explorer/config.yaml
          - --warning
        env:
          - name: EBPF_NET_CLUSTER_NAME
            value: <CLUSTER_NAME>
          - name: EBPF_NET_INTAKE_HOST
            value: RELEASE-NAME-swo-k8s-collector-network-k8s-reducer
          - name: EBPF_NET_INTAKE_PORT
            value: "7000"
          - name: EBPF_NET_CRASH_METRIC_HOST
            value: RELEASE-NAME-swo-k8s-collector-metrics-collector
          - name: EBPF_NET_CRASH_METRIC_PORT
            value: "4317"
        image: otel/opentelemetry-ebpf-k8s-relay:v0.10.1
        imagePullPolicy: IfNotPresent
        name: k8s-relay
        volumeMounts:
          - mountPath: /etc/network-explorer
            name: k8s-relay-config
    initContainers:
      - command:
          - sh
          - -c
          - until nc -zv $EBPF_NET_INTAKE_HOST $EBPF_NET_INTAKE_PORT; do echo "Waiting for reducer endpoint..."; sleep 5; done;
        env:
          - name: EBPF_NET_INTAKE_HOST
            value: RELEASE-NAME-swo-k8s-collector-network-k8s-reducer
          - name: EBPF_NET_INTAKE_PORT
            value: "7000"
        image: busybox:1.36.1
        imagePullPolicy: IfNotPresent
        name: wait-for-reducer
    nodeSelector:
      kubernetes.io/arch: amd64
      kubernetes.io/os: linux
    securityContext: {}
    serviceAccountName: RELEASE-NAME-swo-k8s-collector
    terminationGracePeriodSeconds: 30
    volumes:
      - name: k8s-relay-config
        projected:
          sources:
            - configMap:
                items:
                  - key: config.yaml
                    path: config.yaml
                name: RELEASE-NAME-swo-k8s-collector-network-otel-collector-config
